pkg/sql/alter_primary_key.go

```diff
@@ -129,6 +129,12 @@ func (p *planner) AlterPrimaryKey(
 		"new_primary_key",
 		nameExists,
 	)
+	if alterPKNode.Name != "" &&
+		nameExists(string(alterPKNode.Name)) &&
+		// Allow reuse of existing primary key's name.
+		tableDesc.PrimaryIndex.Name != string(alterPKNode.Name) {
```

Comment:
if you swap the orders of the `nameExists` check and the equality check, it should be slightly better for short-circuiting